src/main/java/gov/usgs/wma/waterdata/collections/BaseController.java

```diff
@@ -17,24 +19,34 @@
 	protected static final String HTTP_500_DESCRIPTION = "Unexpected error occurred.";
 
 	protected void determineContentType(String mimeType) throws HttpMediaTypeNotAcceptableException {
+		determineContentType(mimeType, Arrays.asList(ContentType.values()));
+	}
+
+	protected void determineContentType(String mimeType, List<ContentType> acceptedTypes)
+			throws HttpMediaTypeNotAcceptableException {
 		// TODO: for now just check the query parameter. later 'Accept' header.
 		// Allow for application/json or json as values
-		String baseMimeType = mimeType == null ? null : mimeType.replaceAll(".*/", "");
-		if (baseMimeType == null || baseMimeType.trim().isEmpty() || baseMimeType.equalsIgnoreCase(JSON)) {
-			contentType = JSON;
-		} else if (baseMimeType.equalsIgnoreCase(XML)) {
-			contentType = XML;
-		} else {
-			throw new HttpMediaTypeNotAcceptableException(CONTENT_TYPE_MESS);
+		ContentType contentTypeLocal = null;
+		if (ContentType.json.name().equalsIgnoreCase(mimeType)) {
+			contentTypeLocal = ContentType.json;
+		} else if (ContentType.waterml.name().equalsIgnoreCase(mimeType)) {
+			contentTypeLocal = ContentType.waterml;
+		}
+
+		if (contentTypeLocal == null) {
+			throw new HttpMediaTypeNotAcceptableException(CONTENT_TYPE_MESS + acceptedTypes);
+		} else if(acceptedTypes != null && !acceptedTypes.contains(contentTypeLocal)) {
+			throw new HttpMediaTypeNotAcceptableException(CONTENT_TYPE_MESS + acceptedTypes);
 		}
+		this.contentType = contentTypeLocal;
 	}
 
 	protected boolean contentIsJson() {
-		return JSON.equals(contentType);
+		return ContentType.json.equals(contentType);
 	}
 
-	protected boolean contentIsXml() {
-		return XML.equals(contentType);
+	protected boolean contentIsWaterML() {
+		return ContentType.waterml.equals(contentType);
```

Comment:
I think this would be improved and translate better for the planned upgrade to SB content negotiation if we didn't rely on these two boolean flags.  Instead, have determineContentType return a ContentType (not void) and have the handlers make choices based on that ContentType.